libs/ui/elements/src/lib/metadata-quality/metadata-quality.component.ts

```diff
@@ -69,18 +73,60 @@ export class MetadataQualityComponent implements OnChanges {
       this.items.push({ name, value })
     }
   }
+  hasGetCapabilities(url: string): boolean {
+    const searchParams = new URLSearchParams(url)
+    return searchParams.get('REQUEST')?.toLowerCase() === 'getcapabilities'
+  }
+
+  private readonly COMMON_CHECKS: QualityChecks = {
+    title: (metadata) => !!metadata?.title,
+    description: (metadata) => !!metadata?.abstract,
+    keywords: (metadata) => (metadata?.keywords?.length ?? 0) > 0,
+    legalConstraints: (metadata) =>
+      (metadata?.legalConstraints?.length ?? 0) > 0,
+    contact: (metadata) => !!metadata?.contacts?.[0]?.email,
+  }
+
+  private readonly SPECIFIC_CHECKS: Record<string, QualityChecks> = {
+    dataset: {
+      updateFrequency: (metadata) => !!metadata?.updateFrequency,
+      topic: (metadata) => (metadata?.topics?.length ?? 0) > 0,
+      organisation: (metadata) => !!metadata?.contacts?.[0]?.organization,
+    },
+    service: {
+      capabilities: (metadata) =>
+        (metadata?.onlineResources ?? []).some((resource) =>
+          this.hasGetCapabilities(resource?.url?.search ?? '')
+        ),
+    },
+    reuse: {
+      topic: (metadata) => (metadata?.topics?.length ?? 0) > 0,
+      organisation: (metadata) => !!metadata?.contacts?.[0]?.organization,
+      source: (metadata) => !!metadata?.extras?.sourcesIdentifiers,
+    },
+  }
 
   initialize() {
-    const contact = this.metadata?.contacts?.[0]
     this.items = []
-    this.add('title', !!this.metadata?.title)
-    this.add('description', !!this.metadata?.abstract)
-    this.add('topic', this.metadata?.topics?.length > 0)
-    this.add('keywords', this.metadata?.keywords?.length > 0)
-    this.add('legalConstraints', this.metadata?.legalConstraints?.length > 0)
-    this.add('organisation', !!contact?.organization)
-    this.add('contact', !!contact?.email)
-    this.add('updateFrequency', !!this.metadata?.updateFrequency)
+
+    Object.entries(this.COMMON_CHECKS).forEach(([name, check]) => {
+      this.addIfEnabled(name, check(this.metadata))
+    })
+
+    const datasetType = this.metadata?.kind
+    if (datasetType && this.SPECIFIC_CHECKS[datasetType]) {
+      Object.entries(this.SPECIFIC_CHECKS[datasetType]).forEach(
+        ([name, check]) => {
+          this.addIfEnabled(name, check(this.metadata))
+        }
+      )
+    }
+  }
+
+  private addIfEnabled(name: string, value: boolean) {
```

Comment:
There is still the `add` method with the identical code, that is not used anymore.